### PR TITLE
Excel export for application documents

### PR DESCRIPTION
**Pull request by gakhromov**

### CODE REVIEW

File: back-end/src/lms/views/students.py [hunk 1]
```diff
@@ -176,8 +189,114 @@ def application(self, request: Request, pk=None) -> Response:
...
+                     else 'Нет', center_f),
+                    ('Да' if student.application_process.passport_handed_over
+                     else 'Нет', center_f),
+                    ('Да' if student.application_process.
+                     registration_certificate_handed_over else 'Нет', center_f),
+                    ('Да'
+                     if student.application_process.university_card_handed_over
+                     else 'Нет', center_f),
+                    ('Да' if student.application_process.application_handed_over
+                     else 'Нет', center_f),
```
Comment (TmLev) [inline back-end/src/lms/views/students.py:259]:
```python3
handed_over = []
for handed in ["characteristic_handed_over", ...]:
    handed_over.append("Да" if getattr(student.application_process, handed) else "Нет")
```

File: back-end/src/lms/views/students.py [hunk 2]
```diff
@@ -176,8 +189,114 @@ def application(self, request: Request, pk=None) -> Response:
...
+        Applicants are sorted by campus, specified in request query params.
+        """
+
+        if 'campus' not in request.query_params:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+        campus = request.query_params['campus']
+        students = self.queryset.filter(status='AP',
+                                        university_info__campus=campus)
+        self.generate_excel(students)
+        with open('/tmp/export.xlsx', 'rb') as file:
```
Comment (TmLev) [inline back-end/src/lms/views/students.py:289]: The same name in two places -- could lead to human errors in the future.

File: back-end/src/lms/views/students.py [hunk 3]
```diff
@@ -176,8 +189,114 @@ def application(self, request: Request, pk=None) -> Response:
             data=ApplicationProcessSerializer(instance=updated).data,
         )
 
-
-@extend_schema(tags=["students"])
+    def generate_excel(self, students):
+        """Generate an Excel file with information about the students."""
+
+        workbook = xlsxwriter.Workbook('/tmp/export.xlsx')
```
Comment (TmLev) [inline back-end/src/lms/views/students.py:195]: Why not use https://docs.python.org/3/library/tempfile.html?

File: back-end/src/lms/views/students.py [hunk 4]
```diff
@@ -196,9 +315,9 @@ def list(self, request: Request, *args, **kwargs) -> Response:
         user = request.user
         milgroup = None
 
-        if hasattr(user, "teacher"):
+        if hasattr(user, 'teacher'):
```
Comment (TmLev) [inline back-end/src/lms/views/students.py:318]: :(

File: back-end/src/lms/views/students.py (hunk 2 above)
Comment (gakhromov) [reply back-end/src/lms/views/students.py:289]: That is intended in this case. First, I prepare the variable `students` and then pass it to the method

Comment (TmLev) [reply back-end/src/lms/views/students.py:289]: What? I was talking about the name of the file. More than that, concurrent requests will override this file.

File: back-end/src/lms/views/students.py (hunk 3 above)
Comment (gakhromov) [reply back-end/src/lms/views/students.py:195]: This thing accepts only strings, not file objects https://xlsxwriter.readthedocs.io/workbook.html#constructor

File: back-end/src/lms/views/students.py (hunk 4 above)
Comment (gakhromov) [reply back-end/src/lms/views/students.py:318]: Most of the strings use a single quote, so the linter thing does not allow double ones. I could use double quotes everywhere, but i prefer single ones 😈

Comment (TmLev) [reply back-end/src/lms/views/students.py:318]: :((

File: back-end/src/lms/views/students.py (hunk 1 above)
Comment (gakhromov) [reply back-end/src/lms/views/students.py:259]: Yep, that's much better

File: back-end/src/lms/views/students.py (hunk 2 above)
Comment (gakhromov) [reply back-end/src/lms/views/students.py:289]: Will it really? Can you write on a file that is opened for reading? Thought the `xlsxwriter.Workbook(...)` would wait until the file is not being read anymore

Comment (TmLev) [reply back-end/src/lms/views/students.py:281]: Why would it wait? It's not library's responsibility to watch for such scenarios.
We should probably generate a random name inside of `self.generate_excel` and return it.

Comment (gakhromov) [reply back-end/src/lms/views/students.py:281]: Ok, I will do it this way

**Review by TmLev — APPROVED** [review]
LGTM.

**Pull request merged**